### PR TITLE
fix: nodes unschedulable on Azure

### PR DESCRIPTION
**Pull request by bartusz01**
Nodes cannot be rebooted, because some pods cannot be evicted, which is caused by pdbs (0 allowed disruptions). 
When rebooting the nodes manually this is not a big deal as it requires only 2 extra small steps. However when Kured is installed  (on AKS/ onprem clusters only), the following will happen:

A node is checked by kured and requires a reboot > node is cordoned and drained > some pods cannot be evicted > the node is not rebooted and the node remains cordoned > pods cannot be scheduled on this node anymore.

Normally running 2 replicas where pdb has 1 minAvailable is sufficient. However, knative-serving has a pdb with 80% minAvailable, hence 5 replicas are required to allow 1 pod to be evicted from a node. There is an issue on github for this https://github.com/knative/operator/issues/376.  So this PR is a fix for in the meantime

### CODE REVIEW

**Comment by Morriz** [comment]
Did you check if a new knative release has solved this problem before starting 5 containers where really only one would be enough? It would be preferable to see a good solution

**Review by j-zimnowoda — APPROVED** [review]

**Comment by Morriz** [comment]
Hi Bart, did you check this out: https://github.com/knative/operator/issues/376#issuecomment-973182760
?

**Comment by bartusz01** [comment]
Problem is that we cannot make changes to de PDB during deployment, otherwise we might aswell set ```minAvailable``` to 1.

**Review by 0-sv — APPROVED** [review]

**Comment by j-zimnowoda** [comment]
From the issue:  knative/operator#376 (comment)
> Instead of using minAvailable 80% I use maxUnavailable 20%, which works with 2 replicas
Can we set maxUnavailable via operator ?

**Comment by bartusz01** [comment]
hi @j-zimnowoda Same question like https://github.com/redkubes/otomi-core/pull/663#issuecomment-975550984, so same answer! 😄

**Comment by j-zimnowoda** [comment]
> hi @j-zimnowoda Same question like [#663 (comment)](https://github.com/redkubes/otomi-core/pull/663#issuecomment-975550984), so same answer! 😄

ok, just want to make the question explicit ;) Thanks

File: values/knative/knative-serving-raw.gotmpl [hunk 1]
```diff
@@ -7,6 +7,10 @@ resources:
       namespace: knative-serving
     spec:
       version: 0.23.0
+      high-availability:
+        # Temporary variable #replicas, related to issue: https://github.com/knative/operator/issues/376 
+        # if replicas <5 then pods cannot be evicted (due to pdb) which is a major issue when kured is installed (nodes remain cordoned if pods cannot be evicted!)
+        replicas: {{ $v.charts | get "kured.enabled" "false" | ternary 5 1 }}
```
Comment (Morriz) [inline values/knative/knative-serving-raw.gotmpl:13]:
```suggestion
high-availability:
        # Temporary variable # of replicas, related to issue: https://github.com/knative/operator/issues/376 
        # if replicas <5 then pods cannot be evicted (due to pdb) 
        # TODO: revisit this to check if pub is made  configurable in operator
        replicas: {{ $v.charts | get "kured.enabled" "false" | ternary 5 1 }}
```

File: values/knative/knative-serving-raw.gotmpl [hunk 2]
```diff
@@ -7,6 +7,11 @@ resources:
       namespace: knative-serving
     spec:
       version: 0.23.0
+high-availability:
```
Comment (Morriz) [inline values/knative/knative-serving-raw.gotmpl:10]: here's where the indentation went wrong @j-zimnowoda

Comment (Morriz) [reply values/knative/knative-serving-raw.gotmpl:10]: and @bartusz01 

btw, I set it to 5 always

**Pull request merged**